### PR TITLE
[PDI-20085] - KETTLE_EMPTY_STRING_DIFFERS_FROM_NULL produces opposite display for NULLS - fixing unit tests

### PR DESCRIPTION
**Pull request by andreramos89**
Fixing surrounding unit tests
@pentaho/tatooine_dev

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 24m 46s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 6 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.fieldsplitter.FieldSplitterTest.testSplitFields</b> (click to expand)</summary><p>

```
Unexpected output value at index 1 expected:<null> but was:<>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.fileinput.text.TextFileInputTest.fieldsWithLineBreaksWithEmptyLinesTest</b> (click to expand)</summary><p>

```
[0][0] expected:<null> but was:<>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.fileinput.text.TextFileInputTest.readInputWithNonEmptyNullif</b> (click to expand)</summary><p>

```
expected null, but was:<>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.fileinput.text.TextFileInputTest.readInputWithMissedValues</b> (click to expand)</summary><p>

```
[0][1] expected:<[1]> but was:<[]>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.textfileinput.TextFileInputTest.readInputWithMissedValues</b> (click to expand)</summary><p>

```
[0][1] expected:<[1]> but was:<[]>
```

</p></details>

<details><summary><b>org.pentaho.di.ui.core.dialog.EditRowsDialog_EmptyStringVsNull_Test.emptyAndNullsAreNotDifferent</b> (click to expand)</summary><p>

```
[0][2] expected:<null> but was:<>
```

</p></details>

  
Tests run: **2679**, Failures: **6**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/1/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Review by smmribeiro — APPROVED** [review]
Looks good.

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 24m 33s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 6 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.fieldsplitter.FieldSplitterTest.testSplitFields</b> (click to expand)</summary><p>

```
Unexpected output value at index 1 expected:<null> but was:<>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.fileinput.text.TextFileInputTest.fieldsWithLineBreaksWithEmptyLinesTest</b> (click to expand)</summary><p>

```
[0][0] expected:<null> but was:<>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.fileinput.text.TextFileInputTest.readInputWithNonEmptyNullif</b> (click to expand)</summary><p>

```
expected null, but was:<>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.fileinput.text.TextFileInputTest.readInputWithMissedValues</b> (click to expand)</summary><p>

```
[0][1] expected:<[1]> but was:<[]>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.textfileinput.TextFileInputTest.readInputWithMissedValues</b> (click to expand)</summary><p>

```
[0][1] expected:<[1]> but was:<[]>
```

</p></details>

<details><summary><b>org.pentaho.di.ui.core.dialog.EditRowsDialog_EmptyStringVsNull_Test.emptyAndNullsAreNotDifferent</b> (click to expand)</summary><p>

```
[0][2] expected:<null> but was:<>
```

</p></details>

  
Tests run: **2679**, Failures: **6**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 38m 5s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 6 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.fieldsplitter.FieldSplitterTest.testSplitFields</b> (click to expand)</summary><p>

```
Unexpected output value at index 1 expected:<null> but was:<>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.fileinput.text.TextFileInputTest.fieldsWithLineBreaksWithEmptyLinesTest</b> (click to expand)</summary><p>

```
[0][0] expected:<null> but was:<>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.fileinput.text.TextFileInputTest.readInputWithNonEmptyNullif</b> (click to expand)</summary><p>

```
expected null, but was:<>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.fileinput.text.TextFileInputTest.readInputWithMissedValues</b> (click to expand)</summary><p>

```
[0][1] expected:<[1]> but was:<[]>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.textfileinput.TextFileInputTest.readInputWithMissedValues</b> (click to expand)</summary><p>

```
[0][1] expected:<[1]> but was:<[]>
```

</p></details>

<details><summary><b>org.pentaho.di.ui.core.dialog.EditRowsDialog_EmptyStringVsNull_Test.emptyAndNullsAreNotDifferent</b> (click to expand)</summary><p>

```
[0][2] expected:<null> but was:<>
```

</p></details>

  
Tests run: **5413**, Failures: **6**, Skipped: **15** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/3/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 35m 54s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 2 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.textfileinput.TextFileInputTest.readInputWithMissedValues</b> (click to expand)</summary><p>

```
[0][1] expected:<[1]> but was:<[]>
```

</p></details>

<details><summary><b>org.pentaho.di.ui.core.dialog.EditRowsDialog_EmptyStringVsNull_Test.emptyAndNullsAreNotDifferent</b> (click to expand)</summary><p>

```
[0][2] expected:<null> but was:<>
```

</p></details>

  
Tests run: **2679**, Failures: **2**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/4/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: engine/src/test/java/org/pentaho/di/trans/steps/fileinput/text/TextFileInputTest.java [hunk 1]
```diff
@@ -185,7 +188,7 @@ public void readInputWithMissedValues() throws Exception {
     TextFileInput input = StepMockUtil.getStep( TextFileInput.class, TextFileInputMeta.class, "test" );
     List<Object[]> output = TransTestingUtil.execute( input, meta, data, 2, false );
     TransTestingUtil.assertResult( new Object[] { "1", "1", "1" }, output.get( 0 ) );
-    TransTestingUtil.assertResult( new Object[] { "2", "1", "2" }, output.get( 1 ) );
+    TransTestingUtil.assertResult( new Object[] { "2", "", "2" }, output.get( 1 ) );
```
Comment (smmribeiro) [inline engine/src/test/java/org/pentaho/di/trans/steps/fileinput/text/TextFileInputTest.java:191]: I had another look into this test and the expected is, in fact, to have { "2", **"1"**, "2" } and not { "2", **""**, "2" }!
If we are, now, getting the latter, we are probably introducing a regression somewhere.
I'm removing my previous approval as this whole development seems to need a better analysis.

**Review by smmribeiro — CHANGES_REQUESTED** [review]

File: engine/src/test/java/org/pentaho/di/trans/steps/fileinput/text/TextFileInputTest.java (hunk 1 above)
Comment (andreramos89) [reply engine/src/test/java/org/pentaho/di/trans/steps/fileinput/text/TextFileInputTest.java:191]: This is one of the two tests that are failling.

Comment (andreramos89) [reply engine/src/test/java/org/pentaho/di/trans/steps/fileinput/text/TextFileInputTest.java:191]: I found this comment over the method:
 /*
   * Do not use this method directly! It is for tests!
   */
  @Deprecated
  boolean isNull( Object data, boolean emptyStringDiffersFromNull ) throws KettleValueException {
  
  This method had a wrong comparison with the property also.

**Comment by buildguy** [comment]
### :warning: Build finished in 39m 43s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 2 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.textfileinput.TextFileInputTest.readInputWithMissedValues</b> (click to expand)</summary><p>

```
[0][1] expected:<[1]> but was:<[]>
```

</p></details>

<details><summary><b>org.pentaho.di.ui.core.dialog.EditRowsDialog_EmptyStringVsNull_Test.emptyAndNullsAreNotDifferent</b> (click to expand)</summary><p>

```
[0][2] expected:<null> but was:<>
```

</p></details>

  
Tests run: **2679**, Failures: **2**, Skipped: **5** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/5/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 6m 28s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 4 test(s) failed:

<details><summary><b>org.pentaho.di.core.row.value.ValueMetaStringTest.testCompare_emptyIsNotNull</b> (click to expand)</summary><p>

```
expected:<(<0)> but was:<0>
```

</p></details>

<details><summary><b>org.pentaho.di.core.row.value.ValueMetaStringTest.testIsNull_emptyIsNotNull</b> (click to expand)</summary><p>

```
expected:<false> but was:<true>
```

</p></details>

<details><summary><b>org.pentaho.di.core.row.value.ValueMetaStringTest.testCompare_emptyIsNull</b> (click to expand)</summary><p>

```
expected:<0> but was:<-1>
```

</p></details>

<details><summary><b>org.pentaho.di.core.row.value.ValueMetaStringTest.testIsNull_emptyIsNull</b> (click to expand)</summary><p>

```
expected:<true> but was:<false>
```

</p></details>

  
Tests run: **2000**, Failures: **4**, Skipped: **2** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/6/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 58s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
Command "git checkout -f 314113fa0bc28fbf59842b136d24b8f9bf2c9dfd" returned status code 128:
stdout: 
stderr: fatal: reference is not a tree: 314113fa0bc28fbf59842b136d24b8f9bf2c9dfd
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 26m
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 7 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.ifnull.IfNullTest.testString_emptyIsNull</b> (click to expand)</summary><p>

```
. Unexpected output value at index 2 expected:<replace-value> but was:<>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.ifnull.IfNullTest.testString_emptyIsNotNull</b> (click to expand)</summary><p>

```
. Unexpected output value at index 2 expected:<> but was:<replace-value>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.textfileinput.TextFileInputTest.readInputWithMissedValues</b> (click to expand)</summary><p>

```
[0][1] expected:<[1]> but was:<[]>
```

</p></details>

<details><summary><b>org.pentaho.di.core.row.value.ValueMetaStringTest.testCompare_emptyIsNotNull</b> (click to expand)</summary><p>

```
expected:<(<0)> but was:<0>
```

</p></details>

<details><summary><b>org.pentaho.di.core.row.value.ValueMetaStringTest.testIsNull_emptyIsNotNull</b> (click to expand)</summary><p>

```
expected:<false> but was:<true>
```

</p></details>

<details><summary><b>org.pentaho.di.core.row.value.ValueMetaStringTest.testCompare_emptyIsNull</b> (click to expand)</summary><p>

```
expected:<0> but was:<-1>
```

</p></details>

<details><summary><b>org.pentaho.di.core.row.value.ValueMetaStringTest.testIsNull_emptyIsNull</b> (click to expand)</summary><p>

```
expected:<true> but was:<false>
```

</p></details>

  
Tests run: **3956**, Failures: **7**, Skipped: **6** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/8/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1m 4s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
Command "git checkout -f d2edb1a7bdede5779f22230b337edc13da119fd7" returned status code 128:
stdout: 
stderr: fatal: reference is not a tree: d2edb1a7bdede5779f22230b337edc13da119fd7
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 22m 1s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 3 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.ifnull.IfNullTest.testString_emptyIsNotNull</b> (click to expand)</summary><p>

```
. Unexpected output value at index 2 expected:<> but was:<replace-value>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.ifnull.IfNullTest.testString_emptyIsNull</b> (click to expand)</summary><p>

```
. Unexpected output value at index 2 expected:<replace-value> but was:<>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.textfileinput.TextFileInputTest.readInputWithMissedValues</b> (click to expand)</summary><p>

```
[0][1] expected:<[1]> but was:<[]>
```

</p></details>

  
Tests run: **3956**, Failures: **3**, Skipped: **6** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/10/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 1m 34s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
Command "git checkout -f 314113fa0bc28fbf59842b136d24b8f9bf2c9dfd" returned status code 128:
stdout: 
stderr: fatal: reference is not a tree: 314113fa0bc28fbf59842b136d24b8f9bf2c9dfd
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :warning: Build finished in 28m 41s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 1 test(s) failed:

<details><summary><b>org.pentaho.di.trans.steps.textfileinput.TextFileInputTest.readInputWithMissedValues</b> (click to expand)</summary><p>

```
[0][1] expected:<[1]> but was:<[]>
```

</p></details>

  
Tests run: **3956**, Failures: **1**, Skipped: **6** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/13/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 21m 48s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 5 test(s) failed:

<details><summary><b>org.pentaho.di.core.row.value.ValueMetaBaseTest.testConvertDataFromStringToString</b> (click to expand)</summary><p>

```
KETTLE_DO_NOT_NORMALIZE_SPACES_ONLY_STRING_TO_EMPTY = Y: Conversion from string with spaces only to string must return original string expected:<[]> but was:<[   ]>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.datagrid.DataGrid_EmptyStringVsNull_Test.emptyAndNullsAreDifferent</b> (click to expand)</summary><p>

```

Unexpected conversion error while converting value [number String] to a Number

number String : couldn't convert String to number 

number String : couldn't convert String to number : non-numeric character found at position 1 for value [ ]



```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.fieldsplitter.FieldSplitter_EmptyStringVsNull_Test.emptyAndNullsAreDifferent</b> (click to expand)</summary><p>

```
[0][1] expected:<[]> but was:<[ ]>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.regexeval.RegexEval_EmptyStringVsNull_Test.emptyAndNullsAreDifferent</b> (click to expand)</summary><p>

```
[0][1] expected:<[]> but was:<[ ]>
```

</p></details>

<details><summary><b>org.pentaho.di.ui.core.dialog.EditRowsDialog_EmptyStringVsNull_Test.emptyAndNullsAreDifferent</b> (click to expand)</summary><p>

```
[0][0] expected:<[]> but was:<[ ]>
```

</p></details>

  
Tests run: **3956**, Failures: **5**, Skipped: **6** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/14/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
### :x: Build failed in 25m 3s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,engine,ui
```
---
### :no_entry: Failed Tests

#### :cloud_with_lightning_and_rain: 4 test(s) failed:

<details><summary><b>org.pentaho.di.core.row.value.ValueMetaBaseTest.testConvertDataFromStringToString</b> (click to expand)</summary><p>

```
KETTLE_DO_NOT_NORMALIZE_SPACES_ONLY_STRING_TO_EMPTY = Y: Conversion from string with spaces only to string must return original string expected:<[]> but was:<[   ]>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.datagrid.DataGrid_EmptyStringVsNull_Test.emptyAndNullsAreDifferent</b> (click to expand)</summary><p>

```

Unexpected conversion error while converting value [number String] to a Number

number String : couldn't convert String to number 

number String : couldn't convert String to number : non-numeric character found at position 1 for value [ ]



```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.fieldsplitter.FieldSplitter_EmptyStringVsNull_Test.emptyAndNullsAreDifferent</b> (click to expand)</summary><p>

```
[0][1] expected:<[]> but was:<[ ]>
```

</p></details>

<details><summary><b>org.pentaho.di.trans.steps.regexeval.RegexEval_EmptyStringVsNull_Test.emptyAndNullsAreDifferent</b> (click to expand)</summary><p>

```
[0][1] expected:<[]> but was:<[ ]>
```

</p></details>

  
Tests run: **3956**, Failures: **4**, Skipped: **6** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/15/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 21m 53s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,engine,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **3956**, Failures: **0**, Skipped: **6** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pdi/job/PR-9324/16/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: engine/src/test/java/org/pentaho/di/trans/steps/fileinput/text/TextFileInputTest.java (hunk 1 above)
Comment (andreramos89) [reply engine/src/test/java/org/pentaho/di/trans/steps/fileinput/text/TextFileInputTest.java]: Code reviewed, and test put back to original state

File: core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java [hunk 2]
```diff
@@ -3602,7 +3602,10 @@ boolean isNull( Object data, boolean emptyStringDiffersFromNull ) throws KettleV
         return true;
       }
 
-      if ( emptyStringDiffersFromNull ) {
+      if ( value.equals( StringUtils.EMPTY ) && !emptyStringDiffersFromNull ) {
+        return true;
+      }
+      if ( value.equals( StringUtils.EMPTY ) && emptyStringDiffersFromNull ) {
         return false;
       }
```
Comment (befc) [inline core/src/main/java/org/pentaho/di/core/row/value/ValueMetaBase.java:3610]: I believe these two `if` expressions could be simplified into a direct return of the condition:
```suggestion
      return value.equals(StringUtils.EMPTY) && !emptyStringDiffersFromNull;
```

**Review by befc — APPROVED** [review]
Looks good 👍 

Just added a small comment for a possible condition simplification

**Comment by buildguy** [comment]
### :x: Build failed in 3m 8s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core,engine,ui
```
---
 :exclamation: No tests found!
 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Comment by smmribeiro** [comment]
As requested, the commits were squashed and the development was rebased, however I could not update André's branch, which made me create a new PR ( [pentaho-kettle#9347](https://github.com/pentaho/pentaho-kettle/pull/9347) ).
Closing this one.

**Pull request closed without merging**